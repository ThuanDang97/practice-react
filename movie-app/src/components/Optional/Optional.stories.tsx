import { Story } from '@storybook/react';
import Optional from './Optional';
import { TProps } from './Optional';

export default {
  title: 'Components/Optional',
  component: Optional,
};

const Template: Story<TProps> = (args): JSX.Element => (
  <div
    style={{
      background: 'black',
      position: 'absolute',
      width: '30px',
      height: '30px',
      left: '0',
    }}
  >
    <Optional {...args} />
  </div>
);

export const Basic: Story<TProps> = Template.bind({});
Basic.args = {};
