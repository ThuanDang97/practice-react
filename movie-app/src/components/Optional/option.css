.option {
  position: absolute;
  top: 8px;
  right: 8px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  min-width: 25px;
  min-height: 25px;
  width: 25px;
  height: 25px;
  line-height: inherit;
  background-position: center center;
  background-repeat: no-repeat;
  color: inherit;
  box-sizing: border-box;
  cursor: pointer;
  z-index: 1;
}

.option span {
  background-image: url('https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-947-circle-more-white-4c440dfc1b0e626c70f4853dbbce9c4d1f2c5d8f3e05a7d3df47881cbd816adf.svg');
  font-size: 1.6em;
  position: relative;
  top: 0;
  left: 0;
  display: block;
  align-items: center;
  justify-content: center;
  min-width: 25px;
  min-height: 25px;
  width: 25px;
  height: 25px;
  line-height: inherit;
  background-position: center center;
  background-repeat: no-repeat;
  color: inherit;
  box-sizing: border-box;
}

.option span:hover {
  filter: brightness(0) saturate(100%) invert(53%) sepia(33%) saturate(3054%)
    hue-rotate(156deg) brightness(98%) contrast(99%);
}
