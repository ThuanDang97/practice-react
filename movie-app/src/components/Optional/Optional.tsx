import './option.css';

export type TProps = {
  onHandleClick: (id: number) => void;
  id: number;
};

const Optional = (props: TProps): JSX.Element => {
  const { id, onHandleClick } = props;
  return (
    <div className="option" onClick={() => onHandleClick(id)}>
      <span></span>
    </div>
  );
};

export default Optional;
