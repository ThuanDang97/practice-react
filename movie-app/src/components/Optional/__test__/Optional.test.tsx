import { screen, render } from '@testing-library/react';
import { setup } from '../../../support/Setup';
import Optional, { TProps } from '../Optional';

describe('Test Optional component', () => {
  const defaultProps: TProps = {
    id: 1,
    onHandleClick: jest.fn(),
  };

  test('Component Optional matches DOM Snapshot', () => {
    const component = <Optional {...defaultProps} />;

    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Optional should called handleClick', async () => {
    const { user } = setup(<Optional {...defaultProps} />);

    await user.click(screen.getByTestId('optional'));

    expect(defaultProps.onHandleClick).toHaveBeenCalled();
  });
});
