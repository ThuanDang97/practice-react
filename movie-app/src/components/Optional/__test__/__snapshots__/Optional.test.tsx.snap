// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Optional component Component Optional matches DOM Snapshot 1`] = `
<div>
  <div
    class="option"
    data-testid="optional"
  >
    <span />
  </div>
</div>
`;
